ui/app/decorators/unload-model-route.js

```diff
@@ -0,0 +1,48 @@
+import { inject as service } from '@ember/service';
+import { removeModel } from '../utils/remove-record';
+
+/**
+ * Takes array of model names which should be unloaded when moving away from
+ * the route. Automatically unloads all `capabilities` and `control-group`
+ * data before unloading the specified models.
+ *
+ *** basic example
+ *
+ * import Route from '@ember/route';
+ * import withUnloadModelRoute from 'vault/decorators/unload-model-route';
+ *
+ * @withUnloadModelRoute(['policy/acl'])
+ * class PolicyViewRoute extends Route { foo = null; }
+ *
+ */
+
+export function withUnloadModelRoute(models) {
+  return function decorator(SuperClass) {
+    return class UnloadModelRoute extends SuperClass {
+      @service store;
+
+      static _models;
+
+      constructor() {
+        super(...arguments);
+        if (!models || !Array.isArray(models)) {
+          throw new Error('models must be an array of model types as strings');
+        }
+        this._models = models;
+      }
+
+      unloadModels() {
+        const { _models } = this;
+        _models.forEach((modelName) => {
+          removeModel(this.store, modelName);
+        });
+      }
+
+      deactivate() {
+        ['capabilities', 'control-group', ...this._models].forEach((modelName) => {
```

Comment:
Do you think it will cause any issues forcefully unloading capabilities and control groups?

Reply:
Definitely possible, which is why I'd like to do some heavy testing before we merge something like this. As a fallback we could go through each relationship given model's relationship and reset those (what the helper does) but I think especially if we're only using these on high-level routes, any capabilities and relationships fetched for that section should be unrelated to the new area 🤞